content/blog/highly-available-nfs-workload-on-hpe-greenlake-for-private-cloud-enterprise-using-serviceguard-for-linux.md

```diff
@@ -0,0 +1,748 @@
+         template_id      = data.hpegl_vmaas_template.vanilla.id
+         no_agent         = false
+         asset_tag        = "vm_terraform_sglx"
+         folder_code      = data.hpegl_vmaas_cloud_folder.compute_folder.code
+         create_user      = true
+     }
+ 
+   }
+```
+
+Finally, we will create a VM to act as Serviceguard quorum node:
+
+```jsoniq
+resource "hpegl_vmaas_instance" "my_quorum" {
+     count           = 1 
+     name            = "drbd-${count.index}-qs-${random_integer.random.result}"
+     hostname        = "drbd-${count.index}-qs-${random_integer.random.result}"
+     cloud_id        = data.hpegl_vmaas_cloud.cloud.id
+     group_id        = data.hpegl_vmaas_group.default_group.id
+     layout_id       = data.hpegl_vmaas_layout.vmware.id
+     plan_id         = data.hpegl_vmaas_plan.g2i_medium.id
+     instance_type_code = data.hpegl_vmaas_layout.vmware.instance_type_code
+     network {
+         id = data.hpegl_vmaas_network.blue_segment.id
+     }
+ 
+     volume {
+         name         = "root_vol"
+         size         = 50
+         datastore_id = "auto"
+     }
+
+     config {
+         resource_pool_id = data.hpegl_vmaas_resource_pool.cl_resource_pool.id
+         template_id      = data.hpegl_vmaas_template.vanilla.id
+         no_agent         = false
+         asset_tag        = "vm_terraform_sglx_quorum"
+         folder_code      = data.hpegl_vmaas_cloud_folder.compute_folder.code
+         create_user      = true
+     }
+   }
+```
+
+3 VMs need to be created to setup SGLX. 2 VMs will be used to create Serviceguard for Linux nodes where the NFS service will be up and running. The third VM will act as a quorum server for the Serviceguard cluster to ensure that split brain of the cluster does not impact the availability of the monitored workload.
+
+**Note:** You can get information about each of the resource statements supported by the hpegl provider from [GitHub](https://github.com/hpe/terraform-provider-hpegl/tree/main/docs/resources).
+
+**Note:** An existing Serviceguard Quorum Server in your environment can be used instead of provisioning a third VM, provided the Quorum Server is reachable to the 2 VM’s that were created.
+
+### Terraform init
+
+Before you can use Terraform, you need to initialize it from the configuration file we have created. This is done with the following step: 
+
+`terraform init`
+
+### Terraform ready to plan
+
+To validate your configuration file, it is recommended to run the terraform validate command. Once ready, the terraform plan command will provide the a summary of the deployment that would be built when terraform apply method would be  used.
+Once you agree with the plan and confirm, you can apply the configuration.
+
+### Terraform ready to apply
+
+The command you need to use is now: 
+
+```shell
+terraform apply
+```
+
+ This will rerun the plan command, then prompt you to confirm before it starts building what is in the plan:
```

Comment:
Suggestion for changes:

This will rerun the plan command, then prompt you to confirm before it starts building what is in the plan. Here is some sample output from the `terraform apply` command: